arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/timer.kt

```diff
@@ -80,25 +77,30 @@ suspend fun <A> timeOutOrNull(duration: Duration, fa: suspend () -> A): A? =
     // Register our new tokens to our parents connection
     conn.push(listOf(suspend { timerConn.cancel() }, suspend { faConn.cancel() }))
 
-    suspend { sleep(duration) }.startCoroutineCancellable(
-      CancellableContinuation(
-        EmptyCoroutineContext,
-        timerConn
-      ) { timeOut ->
+    // Launch timer on current thread (Unintercepted) with default ctx (sleep returns there), and timer connection
+    // Launch on current thread, since it will immediately fork to sleeper scheduler, and free current Thread
```

Comment:
I don't get this part, aren't you launching below on the Computation pool thread and not the current one?

Reply:
`suspend { sleep(duration) }` is started with `startCoroutineUnintercepted` which means it immediately starts running without getting `intercepted()` which means it won't get scheduled on` ComputationPool`.

However `sleep` requires a `ContinuationInterceptor` to be installed, otherwise, it cannot shift away from the sleep scheduler when it returns from sleeping. This is problematic since nothing is supposed to run on the scheduling sleep.

So by using `startCoroutineUnintercepted` and `ComputationPool` we can launch the sleeping fiber without first intercepting/scheduling, and we install `ComputationPool` as a `ContinuationInterceptor` for `sleep` to safely return there.